arrow/parser.py

```diff
@@ -327,7 +425,76 @@ def _generate_pattern_re(self, fmt):
 
         return tokens, re.compile(bounded_fmt_pattern, flags=re.IGNORECASE)
 
-    def _parse_token(self, token, value, parts):
+    @overload
+    def _parse_token(
+        self,
+        token: Literal[
+            "YYYY",
+            "YY",
+            "MM",
+            "M",
+            "DDDD",
+            "DDD",
+            "DD",
+            "D",
+            "Do",
+            "HH",
+            "hh",
+            "h",
+            "H",
+            "mm",
+            "m",
+            "ss",
+            "s",
+            "x",
+        ],
+        value: Union[str, bytes, SupportsInt, bytearray],
+        parts: _Parts,
+    ) -> None:
+        ...  # pragma: no cover
+
+    @overload
+    def _parse_token(
+        self,
+        token: Literal["X"],
+        value: Union[str, bytes, SupportsFloat, bytearray],
+        parts: _Parts,
+    ) -> None:
+        ...  # pragma: no cover
+
+    @overload
+    def _parse_token(
+        self,
+        token: Literal["MMMM", "MMM", "dddd", "ddd", "S"],
+        value: Union[str, bytes, bytearray],
+        parts: _Parts,
+    ) -> None:
+        ...  # pragma: no cover
+
+    @overload
+    def _parse_token(
+        self,
+        token: Literal["a", "A", "ZZZ", "ZZ", "Z"],
+        value: Union[str, bytes],
+        parts: _Parts,
+    ) -> None:
+        ...  # pragma: no cover
+
+    @overload
+    def _parse_token(
+        self,
+        token: Literal["W"],
+        value: Tuple[_WEEKDATE_ELEMENT, _WEEKDATE_ELEMENT, Optional[_WEEKDATE_ELEMENT]],
+        parts: _Parts,
+    ) -> None:
+        ...  # pragma: no cover
+
+    def _parse_token(
+        self,
+        token: Any,
+        value: Any,
+        parts: Any,
+    ) -> None:
```

Comment:
Mind explaining the rationale for `value`'s type in these overloads?

From this:
https://github.com/arrow-py/arrow/blob/ea9f325dea89194fbd07cf74ad34a7e95f65ae4f/arrow/parser.py#L235-L248

It seems like the expected types for `value` should be `value: Union[Tuple[str, str, str], str]`

Reply:
@jadchaar  Ah, when I add type annotation on existing code, I do not consider caller side's input and always try to consider inside of the function. because it helps me to narrow the signature down.
But it could be too much for internal purpose method. Do you want me to simplify this?